extension/awsproxy/factory_test.go

```diff
@@ -81,10 +82,14 @@ func TestFactory_CreateExtension(t *testing.T) {
 	err = ext.Start(ctx, mh)
 	assert.NoError(t, err)
 
-	resp, err := http.Post(
-		"http://"+address+"/GetSamplingRules",
-		"application/json",
-		strings.NewReader(`{"NextToken": null}`))
+	var resp *http.Response
+	assert.Eventually(t, func() bool {
```

Comment:
```suggestion
require.Eventually(t, func() bool {
```

If this fails, the remaining tests will panic due to `resp` nil

Reply:
Good catch, thanks!